newrelic/hooks/mlmodel_openai.py

```diff
@@ -54,46 +53,75 @@ def wrap_embedding_create(wrapped, instance, args, kwargs):
     if not transaction:
         return wrapped(*args, **kwargs)
 
+    # Framework metric also used for entity tagging in the UI
     transaction.add_ml_model_info("OpenAI", OPENAI_VERSION)
 
+    # Obtain attributes to be stored on embedding events regardless of whether we hit an error
+    request_args = kwargs
+    embedding_id = str(uuid.uuid4())
+
+    # Get API key without using the response so we can store it before the response is returned in case of errors
+    api_key = getattr(openai, "api_key", None)
+    api_key_last_four_digits = f"sk-{api_key[-4:]}" if api_key else ""
+
+    # Get trace information
+    available_metadata = get_trace_linking_metadata()
+    span_id = available_metadata.get("span.id", "")
+    trace_id = available_metadata.get("trace.id", "")
+
+    settings = transaction.settings if transaction.settings is not None else global_settings()
+
+    # Shared attributes that will be included on all embedding events (error or not)
+    base_embedding_dict = {
+        "embedding_id": embedding_id,
+        "appName": settings.app_name,
+        "api_key_last_four_digits": api_key_last_four_digits,
+        "span_id": span_id,
```

Comment:
Is there a bug in this code right now where we are capturing the wrong span id? Shouldn't we be capturing the span id inside the function trace? I just looked at the SDK code and they attach the function trace's span id.

Reply:
I'm not sure how this would work in practice but what I was imaging for this design and what I would propose is that this code wouldn't change this much from the previous version and that our .get(attr_name, "") fallback logic would handle most of the cases where things don't exist because an error occurred. I'd propose maybe approaching the implementation from that perspective and see if that works out better?